gui/README.md

```diff
@@ -0,0 +1,40 @@
+# GUI system
+
+This directory contains the XML files describing the components which make up Graphite's GUI.
+
+## Principles
+
+The framework is inspired by [Vue.js](https://vuejs.org/).
+Each component's layout is defined in an XML, and recursively made out of lower-level components.
+
+Interactivity is provided by script files which expose reactive variables. As these variables are mutated, the component is updated to match the current state.
+
+## Layout
+
+The layout engine does a top-down pass through the component tree in order to determine what to render.
+
+Layout is controlled using predefined attributes, such as `width`, `height`, `x-align`, `y-align`, `x-padding`, `y-padding`.
+
+## Component lifetime
+
+The children of a component are passed to it as a `content` attribute. For example, looking at the row component:
+```xml
+<row content="INNER_XML: (None) = none">
+    {{INNER_XML}}
+</row>
+```
+The `content` attribute defines a new variable `INNER_XML`, of any type, and with no default value.
```

Comment:
The `content` attribute defines a new variable `INNER_XML`, of either the GuiXml or None, and with no default value.